### ISSUE TITLE
tmkms yubihsm setup deletes default authentication key - how to find new admin password after running tmkms setup?

### ISSUE THREAD

**Issue by kobrien** [iqlusioninc/tmkms#371]
I am after running `tmkms yubihsm setup` on a brand new yubihsm device, I now no longer have access to the default authentication key (password=password), nor can I find the new admin password anywhere.

according to the docs here https://github.com/iqlusioninc/tmkms/blob/main/README.yubihsm.md

```
NOTE: the admin password is displayed on two separate lines. When using it from yubihsm-shell or as the password field in tmkms.toml, it is not split across multiple lines and is separated only by a single space between words.
```

I dont see the admin password anywhere in my terminal output, where can I find this?

I need to perform a factory reset, using yubihsm-shell, but i no longer have admin access to the device due to running `tmkms setup` which has deleted the default authentication key

**Comment by tony-iqlusion** [comment]
To answer your immediate question: you'll need to perform a full reset of the YubiHSM2. The procedure for that is to plug it in and, immediately after doing so, push on it and continue to do so for 10 seconds (the top is effectively a button). That will perform a full factory reset.

We've never had someone report that the init command does not print the passwords. Can you provide a full log of what happened?

**Comment by kobrien** [comment]
Hi Tony, the issue I'm having is with the setup command, it deletes the authentication key when ran, and creates the 4 new keys (identity, validator etc), I can no longer run tmkms setup again (say to create a new set of keys), because I no longer have the admin password

```
output from running tmkms setup
*** Are you SURE you want erase and reinitialize this HSM? (y/N): y
2021-07-23T18:01:48.211305Z  WARN yubihsm::client: factory resetting HSM device! all data will be lost!
2021-07-23T18:01:49.259750Z  INFO yubihsm::client: waiting for device reset to complete
2021-07-23T18:01:51.051613Z  INFO yubihsm::setup: installed temporary setup authentication key into slot 65534


2021-07-23T18:01:51.070833Z  WARN yubihsm::setup: deleting default authentication key from slot 1


2021-07-23T18:01:51.096218Z  INFO yubihsm::setup::profile: installing role: admin:2021-07-23T18:01:46Z
2021-07-23T18:01:51.119015Z  INFO yubihsm::setup::profile: installing role: operator:2021-07-23T18:01:46Z
2021-07-23T18:01:51.138391Z  INFO yubihsm::setup::profile: installing role: auditor:2021-07-23T18:01:46Z
2021-07-23T18:01:51.157715Z  INFO yubihsm::setup::profile: installing role: validator:2021-07-23T18:01:46Z
2021-07-23T18:01:51.177037Z  INFO yubihsm::setup::profile: installing wrap key: primary:2021-07-23T18:01:46Z
2021-07-23T18:01:51.199035Z  INFO yubihsm::setup::profile: storing provisioning report in opaque object 0xfffe
2021-07-23T18:01:51.223016Z  WARN yubihsm::setup: deleting temporary setup authentication key from slot 65534
     Success reinitialized YubiHSM (serial: 0016499241)
```

**Comment by kobrien** [comment]
anybody able to help shed some light on this?

**Comment by tony-iqlusion** [comment]
I already explained above how to do a factory reset, but let me try again step-by-step:

1. Insert the YubiHSM2 into a USB port
2. *Immediately* after inserting it, continue to press on the top for 10 seconds (it's like a button)
3. The YubiHSM2 will factory reset itself, reverting back to the default admin password (you may need to remove it and insert it after that, I don't recall)

**Comment by kobrien** [comment]
My issue isn't with factory resetting the device. The problem is with running tmkms setup, this deletes the admin auth key. I would like to run tmkms setup, in order to generate the various keys, but keep the admin auth key.

**Comment by tony-iqlusion** [comment]
There's presently no option to do that without provisioning an auth key, sorry.

Are you still unable to get it to print out the keys it provisions? That's strange.

**Comment by kobrien** [comment]
Hello, sorry for the delay, I ran `tmkms setup` on the yubikey and as I predicted the mneumonic does not work as the admin password. I no longer have shell access to the device using yubihsm-shell. 

```
yubihsm> session open 1 <mnemonic>
Failed to create session: Unable to verify cryptogram
```

**Comment by kobrien** [comment]
ok this is very strange, the mneumonic seems to work when I add it to the tmkms config, as I can run `keys list` without getting an authentication error, however it tells me there are no keys on the device? I literally just created a set of keys, using 
`setup`.

```
tmkms yubihsm keys list -c /config/tmkms.toml
error: no keys in this YubiHSM (#########)
```

**Comment by kobrien** [comment]
re-reading the README here, yes I have removed the multiple line from the mnemonic, the words are separated by spaces
https://github.com/iqlusioninc/tmkms/blob/main/README.yubihsm.md
```
NOTE: the admin password is displayed on two separate lines. When using it from yubihsm-shell or as the password field in tmkms.toml, it is not split across multiple lines and is separated only by a single space between words.
```

**Comment by kobrien** [comment]
oh wow i think the issue was missing double quotes......

**Comment by kobrien** [comment]
closing as issue was resolved by wrapping the mneumonic in double quotes, when passing to yubihsm shell
